src/python/pants/init/engine_initializer.py

```diff
@@ -171,66 +170,53 @@ def setup_legacy_graph(
-            bootstrap_options.local_execution_root_dir,
-            bootstrap_options.named_caches_dir,
-            bootstrap_options.build_file_prelude_globs,
             options_bootstrapper,
             build_configuration,
+            ExecutionOptions.from_bootstrap_options(bootstrap_options),
+            pants_ignore_patterns=OptionsInitializer.compute_pants_ignore(
+                build_root, bootstrap_options
+            ),
+            use_gitignore=use_gitignore,
+            local_store_dir=bootstrap_options.local_store_dir,
+            local_execution_root_dir=bootstrap_options.local_execution_root_dir,
+            named_caches_dir=bootstrap_options.named_caches_dir,
             build_root=build_root,
             native=native,
             glob_match_error_behavior=(
                 bootstrap_options.files_not_found_behavior.to_glob_match_error_behavior()
             ),
+            build_file_prelude_globs=bootstrap_options.build_file_prelude_globs,
+            build_patterns=bootstrap_options.build_patterns,
             build_ignore_patterns=bootstrap_options.build_ignore,
+            tags=bootstrap_options.tag,
             exclude_target_regexps=bootstrap_options.exclude_target_regexp,
             subproject_roots=bootstrap_options.subproject_roots,
             include_trace_on_error=bootstrap_options.print_exception_stacktrace,
-            execution_options=ExecutionOptions.from_bootstrap_options(bootstrap_options),
         )
 
     @staticmethod
     def setup_legacy_graph_extended(
+        options_bootstrapper: OptionsBootstrapper,
+        build_configuration: BuildConfiguration,
+        execution_options: ExecutionOptions,
+        *,
         pants_ignore_patterns: List[str],
         use_gitignore: bool,
         local_store_dir: str,
         local_execution_root_dir: str,
         named_caches_dir: str,
-        build_file_prelude_globs: Tuple[str, ...],
-        options_bootstrapper: OptionsBootstrapper,
-        build_configuration: BuildConfiguration,
-        execution_options: ExecutionOptions,
         build_root: Optional[str] = None,
         native: Optional[Native] = None,
         glob_match_error_behavior: GlobMatchErrorBehavior = GlobMatchErrorBehavior.warn,
-        build_ignore_patterns=None,
-        exclude_target_regexps=None,
-        subproject_roots=None,
+        build_patterns: Optional[Iterable[str]] = None,
+        build_file_prelude_globs: Optional[Iterable[str]] = None,
+        build_ignore_patterns: Optional[Iterable[str]] = None,
+        tags: Optional[Iterable[str]] = None,
+        exclude_target_regexps: Optional[Iterable[str]] = None,
+        subproject_roots: Optional[Iterable[str]] = None,
         include_trace_on_error: bool = True,
     ) -> LegacyGraphScheduler:
-        """Construct and return the components necessary for LegacyBuildGraph construction.
-
-        :param local_store_dir: The directory to use for storing the engine's LMDB store in.
```

Comment:
We were missing docstring on a couple of the parameters. Rather than updating, I figured we aren't getting much value from them because the type hints and names are already fairly self-explanatory.